### ISSUE TITLE
Multiple issues after upgrading deb-get version > 0.3.6-1

### ISSUE THREAD

**Issue by bcook128** [wimpysworld/deb-get#1052]
# deb-get issue template

Thank you for taking the time to report an issue to the `deb-get` project 🙂

 - Please complete one the appropriate sections below.
 - Delete all the sections that do not apply, including this one.
 - Any issues that do not use this template will be closed.
 - If you're not sure what to do, or have a general question about `deb-get`, then ask in [Wimpy's World Discord](https://wimpysworld.io/discord).

## Reporting a bug 🐛

Please make sure the version of `deb-get` you are using is up to date:

```bash
deb-get update
deb-get reinstall deb-get
```

### Expected behavior

When running "deb-get list", output should show all available apps and indicate if an app is installed "[Installed]".

Output of "deb-get list" for deb-get version 0.3.6-1:
```bash
1password
android-messages-desktop
antimicrox
anydesk
appimagelauncher
atom
audio-recorder
azure-cli
azuredatastudio
balena-etcher-electron
bat                             [ installed ]
battery-monitor
beersmith3
bitwarden
bitwig-studio
blanket
blockbench
bluejeans-v2
bottom                          [ installed ]
brave-browser
brisqi
caprine
cawbird
chronograf
code
codium
com.github.tkashkin.gamehub
copyq
crossover
cryptomator
dbeaver-ce
deadbeef
deb-get                         [ installed ]
deborah
deltachat-desktop
discord
docker-ce
docker-desktop
draw.io
dropbox
du-dust                         [ installed ]
duf                             [ installed ]
dustracing2d
element-desktop
enpass
exodus
expressvpn
fd                              [ installed ]
figma-linux
firefox-esr
flameshot
foliate
franz
fsearch                         [ installed ]
geforcenow-electron
gh                              [ installed ]
git-delta                       [ installed ]
github-desktop
gitkraken
gitter
google-chat-electron
google-chrome-stable            [ installed ]
google-cloud-cli
google-earth-pro-stable         [ installed ]
gpu-viewer
grype
helio-workstation
heroic
hugo
hyper
igdm
influxdb
influxdb2
influxdb2-cli
insomnia
insync
irccloud-desktop
jabref
jami
jellyfin
kapacitor
kdiskmark
keepassxc
keybase
kopia-ui
ksnip
lens
librewolf
lsd                             [ installed ]
ludo
lutris
mailspring                      [ installed ]
mattermost-desktop
mergerfs
micro                           [ installed ]
microsoft-edge-stable
min
minigalaxy
minikube
motrix
mpdevil
ms-office-electron
mullvad-vpn
neo4j
neovim                          [ installed ]
nextcloud-desktop
nomad
nordvpn
notable
obs-cli
obsidian                        [ installed ]
obs-studio
ocenaudio
onlyoffice-desktopeditors
openaudible                     [ installed ]
openrazer-meta
openrgb
opera-stable
p3x-onenote
pandoc                          [ installed ]
papirus-icon-theme
parsec
peazip
picocrypt
plexmediaserver
plow
polychromatic
portmaster
powershell
protonmail-bridge
protonvpn
pulse-sms
qownnotes
quickemu                        [ installed ]
quickgui                        [ installed ]
rambox
rclone                          [ installed ]
resilio-sync
retroarch
rocketchat
rpi-imager                      [ installed ]
rstudio
rustdesk
sejda-desktop
shutter-encoder
signal-desktop
simplenote
skypeforlinux
slack-desktop
sleek
smartgit
softmaker-office-2021
soundux
spotify-client
standard-notes
strawberry
stremio
sublime-merge
sublime-text
surfshark
syft
syncthing
system-monitoring-center
tailscale                       [ installed ]
teams
teams-for-linux
teamviewer                      [ installed ]
telegraf
terraform
texworks
tidal-hifi
tixati
tribler
trivy
typora
ubuntu-make
ulauncher                       [ installed ]
virtualbox-6.1
vivaldi-stable
vuescan
wavebox
waydroid
webex
weechat
weylus
whalebird
whatsapp-for-linux
wire-desktop
xemu
youtube-music
yq
zenith                          [ installed ]
zettlr
zoom
zotero                          [ installed ]
```

### Actual behavior

After upgrading deb-get to any version later than 03.6-1, running command "deb-get list" produces errors stating that caches for apps are empty and usually hangs midway through the list, and exits when any key is pressed. This behavior is consistent across multiple machines running same OS (Ubuntu 22.04) and has occurred since 0.3.6-1 was not latest release. 

deb-get clean does not fix.

DEBGET_TOKEN variable is set with a valid Github PAT on all machines experiencing the issue.
```bash
echo $DEBGET_TOKEN
github_pat_***************************************************************************************
```

Output when running deb-get version > 0.3.6-1:
```bash
1password
activitywatch
agena
alduin
android-messages-desktop
antimicrox
anydesk
appimagelauncher
atom
atomic
  [*] WARNING! Cached file /var/cache/deb-get/atuin.json is empty or missing.
atuin
audio-recorder
  [*] WARNING! Cached file /var/cache/deb-get/authme.json is empty or missing.
authme
azure-cli
  [*] WARNING! Cached file /var/cache/deb-get/azuredatastudio.json is empty or missing.
azuredatastudio
  [*] WARNING! Cached file /var/cache/deb-get/balena-etcher.json is empty or missing.
balena-etcher
bat
battery-monitor
beersmith3
  [*] WARNING! Cached file /var/cache/deb-get/bitwarden.json is empty or missing.
bitwarden
bitwig-studio
blanket
  [*] WARNING! Cached file /var/cache/deb-get/blockbench.json is empty or missing.
blockbench
bluejeans-v2
bottom
brave-browser
  [*] WARNING! Cached file /var/cache/deb-get/brisqi.json is empty or missing.
brisqi
  [*] WARNING! Cached file /var/cache/deb-get/browsh.json is empty or missing.
browsh
  [*] WARNING! Cached file /var/cache/deb-get/caddy.json is empty or missing.
caddy
  [*] WARNING! Cached file /var/cache/deb-get/caprine.json is empty or missing.
caprine
cawbird
chronograf
code
  [*] WARNING! Cached file /var/cache/deb-get/codium.json is empty or missing.
codium
  [*] WARNING! Cached file /var/cache/deb-get/com.github.tkashkin.gamehub.json is empty or missing.
com.github.tkashkin.gamehub
copyq
  [*] WARNING! Cached file /var/cache/deb-get/cosign.json is empty or missing.
cosign
  [*] WARNING! Cached file /var/cache/deb-get/croc.json is empty or missing.
croc
cronopete
crossover
  [*] WARNING! Cached file /var/cache/deb-get/crow-translate.json is empty or missing.
crow-translate
cryptomator
dbeaver-ce
deadbeef-static
deb-get                         [ installed ]
  [*] WARNING! Cached file /var/cache/deb-get/deborah.json is empty or missing.
deborah
deltachat-desktop
  [*] WARNING! Cached file /var/cache/deb-get/deskcut.json is empty or missing.
deskcut
  [*] WARNING! Cached file /var/cache/deb-get/detectiteasy.json is empty or missing.
detectiteasy
discord
  [*] WARNING! Cached file /var/cache/deb-get/dive.json is empty or missing.
dive
docker-ce
docker-desktop
doublecmd-gtk
doublecmd-qt
  [*] WARNING! Cached file /var/cache/deb-get/draw.io.json is empty or missing.
draw.io
dropbox
du-dust
duf
  [*] WARNING! Cached file /var/cache/deb-get/dustracing2d.json is empty or missing.
dustracing2d
element-desktop
  [*] WARNING! Cached file /var/cache/deb-get/emby-server.json is empty or missing.
emby-server
enpass
  [*] WARNING! Cached file /var/cache/deb-get/exifcleaner.json is empty or missing.
exifcleaner
expressvpn
  [*] WARNING! Cached file /var/cache/deb-get/fail2ban.json is empty or missing.
fail2ban
  [*] WARNING! Cached file /var/cache/deb-get/fastfetch.json is empty or missing.
fastfetch
fd
  [*] WARNING! Cached file /var/cache/deb-get/ferdium.json is empty or missing.
ferdium
  [*] WARNING! Cached file /var/cache/deb-get/figma-linux.json is empty or missing.
figma-linux
filezilla-theme-papirus
firefox
firefox-beta
firefox-devedition
firefox-esr
firefox-nightly
  [*] WARNING! Cached file /var/cache/deb-get/flameshot.json is empty or missing.
flameshot
foliate
  [*] WARNING! Cached file /var/cache/deb-get/franz.json is empty or missing.
franz
  [*] WARNING! Cached file /var/cache/deb-get/frostwire.json is empty or missing.
frostwire
fsearch
  [*] WARNING! Cached file /var/cache/deb-get/gcm.json is empty or missing.
gcm
gh
git-delta
  [*] WARNING! Cached file /var/cache/deb-get/github-desktop.json is empty or missing.
github-desktop
gitkraken
gitter
glow
  [*] WARNING! Cached file /var/cache/deb-get/google-chat-electron.json is empty or missing.
google-chat-electron
google-chrome-beta
google-chrome-stable
google-chrome-unstable
google-cloud-cli
google-earth-pro-stable
gpu-viewer
grub-customizer
  [*] WARNING! Cached file /var/cache/deb-get/grype.json is empty or missing.
grype
headset
helio-workstation
  [*] WARNING! Cached file /var/cache/deb-get/heroic.json is empty or missing.
heroic
  [*] WARNING! Cached file /var/cache/deb-get/homeassistant-supervised.json is empty or missing.
homeassistant-supervised
  [*] WARNING! Cached file /var/cache/deb-get/hugo.json is empty or missing.
hugo
  [*] WARNING! Cached file /var/cache/deb-get/hyper.json is empty or missing.
hyper
  [*] WARNING! Cached file /var/cache/deb-get/hyperfine.json is empty or missing.
hyperfine
i3
  [*] WARNING! Cached file /var/cache/deb-get/igdm.json is empty or missing.
igdm
  [*] WARNING! Cached file /var/cache/deb-get/imhex.json is empty or missing.
imhex
indicator-sound-switcher
influxdb
influxdb2
influxdb2-cli
inkscape
  [*] WARNING! Cached file /var/cache/deb-get/insomnia.json is empty or missing.
insomnia
insync
  [*] WARNING! Cached file /var/cache/deb-get/ipfs-desktop.json is empty or missing.
ipfs-desktop
  [*] WARNING! Cached file /var/cache/deb-get/irccloud-desktop.json is empty or missing.
irccloud-desktop
iriunwebcam
  [*] WARNING! Cached file /var/cache/deb-get/jabref.json is empty or missing.
jabref
jami
  [*] WARNING! Cached file /var/cache/deb-get/jc.json is empty or missing.
jc
jellyfin
  [*] WARNING! Cached file /var/cache/deb-get/jellyfin-media-player.json is empty or missing.
jellyfin-media-player
  [*] WARNING! Cached file /var/cache/deb-get/jexiftoolgui.json is empty or missing.
jexiftoolgui
  [*] WARNING! Cached file /var/cache/deb-get/jupyterlab-desktop.json is empty or missing.
jupyterlab-desktop
kapacitor
kdiskmark
keepassxc
keybase
kopia-ui
  [*] WARNING! Cached file /var/cache/deb-get/koreader.json is empty or missing.
koreader
  [*] WARNING! Cached file /var/cache/deb-get/kse.json is empty or missing.
kse
  [*] WARNING! Cached file /var/cache/deb-get/ksnip.json is empty or missing.
ksnip
kwaterfoxhelper
lens
libreoffice
librewolf
  [*] WARNING! Cached file /var/cache/deb-get/limoji.json is empty or missing.
limoji
  [*] WARNING! Cached file /var/cache/deb-get/localsend.json is empty or missing.
localsend
lsd
  [*] WARNING! Cached file /var/cache/deb-get/ludo.json is empty or missing.
ludo
  [*] WARNING! Cached file /var/cache/deb-get/lutris.json is empty or missing.
lutris
mailspring
makedeb
master-pdf-editor-5
  [*] WARNING! Cached file /var/cache/deb-get/mattermost-desktop.json is empty or missing.
mattermost-desktop
  [*] WARNING! Cached file /var/cache/deb-get/mdview.json is empty or missing.
mdview
media-downloader
mediathekview
  [*] WARNING! Cached file /var/cache/deb-get/mergerfs.json is empty or missing.
mergerfs
micro
microsoft-edge-stable
  [*] WARNING! Cached file /var/cache/deb-get/min.json is empty or missing.
min
  [*] WARNING! Cached file /var/cache/deb-get/minigalaxy.json is empty or missing.
minigalaxy
  [*] WARNING! Cached file /var/cache/deb-get/minikube.json is empty or missing.
minikube
  [*] WARNING! Cached file /var/cache/deb-get/motrix.json is empty or missing.
motrix
  [*] WARNING! Cached file /var/cache/deb-get/mpdevil.json is empty or missing.
mpdevil
  [*] WARNING! Cached file /var/cache/deb-get/ms-365-electron.json is empty or missing.
ms-365-electron
  [*] WARNING! Cached file /var/cache/deb-get/mullvad-vpn.json is empty or missing.
mullvad-vpn
multimc
nala
  [*] WARNING! Cached file /var/cache/deb-get/nekoray.json is empty or missing.
nekoray
  [*] WARNING! Cached file /var/cache/deb-get/nemo-mediainfo-tab.json is empty or missing.
nemo-mediainfo-tab
neo4j
nextcloud-desktop
nodejs
nomad
nordvpn
  [*] WARNING! Cached file /var/cache/deb-get/notable.json is empty or missing.
notable
  [*] WARNING! Cached file /var/cache/deb-get/nuclear.json is empty or missing.
nuclear
  [*] WARNING! Cached file /var/cache/deb-get/obs-cli.json is empty or missing.
obs-cli
obsidian
obs-studio
ocenaudio
onedriver
  [*] WARNING! Cached file /var/cache/deb-get/onlyoffice-desktopeditors.json is empty or missing.
onlyoffice-desktopeditors
openaudible
openrazer-meta
openrgb
opera-stable
  [*] WARNING! Cached file /var/cache/deb-get/os-agent.json is empty or missing.
os-agent
  [*] WARNING! Cached file /var/cache/deb-get/p3x-onenote.json is empty or missing.
p3x-onenote
pandoc
papirus-folders
papirus-icon-theme
  [*] WARNING! Cached file /var/cache/deb-get/pdfsam-basic.json is empty or missing.
pdfsam-basic
  [*] WARNING! Cached file /var/cache/deb-get/pdfsam-visual.json is empty or missing.
pdfsam-visual
  [*] WARNING! Cached file /var/cache/deb-get/peazip.json is empty or missing.
peazip
plexmediaserver
  [*] WARNING! Cached file /var/cache/deb-get/plow.json is empty or missing.
plow
polychromatic
  [*] WARNING! Cached file /var/cache/deb-get/pomatez.json is empty or missing.
pomatez
  [*] WARNING! Cached file /var/cache/deb-get/popcorn-time.json is empty or missing.
popcorn-time
portmaster
  [*] WARNING! Cached file /var/cache/deb-get/powershell.json is empty or missing.
powershell
  [*] WARNING! Cached file /var/cache/deb-get/protonmail-bridge.json is empty or missing.
protonmail-bridge
protonvpn
publii
  [*] WARNING! Cached file /var/cache/deb-get/pulsar.json is empty or missing.
pulsar
  [*] WARNING! Cached file /var/cache/deb-get/pulse-sms.json is empty or missing.
pulse-sms
qownnotes
  [*] WARNING! Cached file /var/cache/deb-get/quarto.json is empty or missing.
quarto
quickemu
quickgui
  [*] WARNING! Cached file /var/cache/deb-get/quickobs.json is empty or missing.
quickobs
  [*] WARNING! Cached file /var/cache/deb-get/rambox.json is empty or missing.
rambox
rclone
resilio-sync
retroarch
  [*] WARNING! Cached file /var/cache/deb-get/ripgrep.json is empty or missing.
ripgrep
  [*] WARNING! Cached file /var/cache/deb-get/rocketchat.json is empty or missing.
rocketchat
rpi-imager
rstudio
rstudio-server
  [*] WARNING! Cached file /var/cache/deb-get/rustdesk.json is empty or missing.
rustdesk
sejda-desktop
shutter-encoder
signal-desktop
  [*] WARNING! Cached file /var/cache/deb-get/simplenote.json is empty or missing.
simplenote
simple-pwgen
skypeforlinux
slack-desktop
  [*] WARNING! Cached file /var/cache/deb-get/sleek.json is empty or missing.
sleek
smartgit
  [*] WARNING! Cached file /var/cache/deb-get/sniffnet.json is empty or missing.
sniffnet
softmaker-office-2021
songrec
  [*] WARNING! Cached file /var/cache/deb-get/soundux.json is empty or missing.
soundux
speedtest
spotify-client
  [*] WARNING! Cached file /var/cache/deb-get/standard-notes.json is empty or missing.
standard-notes
strawberry
sublime-merge
sublime-text
  [*] WARNING! Cached file /var/cache/deb-get/sunshine.json is empty or missing.
sunshine
surfshark
  [*] WARNING! Cached file /var/cache/deb-get/syft.json is empty or missing.
syft
syncthing
  [*] WARNING! Cached file /var/cache/deb-get/system-monitoring-center.json is empty or missing.
system-monitoring-center
  [*] WARNING! Cached file /var/cache/deb-get/tabby-terminal.json is empty or missing.
tabby-terminal
tailscale
tarsnap
teams
  [*] WARNING! Cached file /var/cache/deb-get/teams-for-linux.json is empty or missing.
teams-for-linux
teamviewer
  [*] WARNING! Cached file /var/cache/deb-get/teip.json is empty or missing.
teip
telegraf
termius-app
terraform
texworks
tezos-baking
thinlinc-client
  [*] WARNING! Cached file /var/cache/deb-get/thorium-browser.json is empty or missing.
thorium-browser
  [*] WARNING! Cached file /var/cache/deb-get/tidal-hifi.json is empty or missing.
tidal-hifi
tixati
tofu
tonelib-bassdrive
tonelib-gfx
tonelib-jam
tonelib-metal
tonelib-noisereducer
tonelib-tubewarmth
tonelib-zoom
  [*] WARNING! Cached file /var/cache/deb-get/tribler.json is empty or missing.
tribler
  [*] WARNING! Cached file /var/cache/deb-get/trilium.json is empty or missing.
trilium
  [*] WARNING! Cached file /var/cache/deb-get/trivy.json is empty or missing.
trivy
typora
ubuntu-make
ulauncher
urserver
usbimager
vagrant
veracrypt
virtualbox-6.1
virtualbox-7.0
vivaldi-stable
vuescan
waterfox-g-kde
wavebox
waydroid
webex
weechat
  [*] WARNING! Cached file /var/cache/deb-get/weylus.json is empty or missing.
weylus
  [*] WARNING! Cached file /var/cache/deb-get/whalebird.json is empty or missing.
whalebird
  [*] WARNING! Cached file /var/cache/deb-get/whatsapp-for-linux.json is empty or missing.
whatsapp-for-linux
winehq-devel
winehq-stable
winehq-staging
wire-desktop
  [*] WARNING! Cached file /var/cache/deb-get/wraith-master.json is empty or missing.
wraith-master
xemu
  [*] WARNING! Cached file /var/cache/deb-get/youtube-music.json is empty or missing.
youtube-music
yq
zenith
  [*] WARNING! Cached file /var/cache/deb-get/zettlr.json is empty or missing.
zettlr
zoom
zotero


```
### Steps to reproduce

```bash
deb-get list
```

### System information

Run the following a paste the output below:

#### /etc/os-release

```bash
PRETTY_NAME="Ubuntu 22.04.4 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.4 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```
#### /etc/lsb-release

```bash
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.4 LTS"

```

#### deb-get version

```bash
0.4.0
```

### deb-get update

```bash
  [+] Updating /etc/deb-get/01-main
  -=O=-          #          #           #            #                                                                                                                                                           
  [+] Updating cache of supported apps in the background
Get:1 http://security.ubuntu.com/ubuntu jammy-security InRelease [110 kB]
Hit:2 http://dl.google.com/linux/earth/deb stable InRelease
Hit:3 http://deb.volian.org/volian scar InRelease
Hit:4 http://us.archive.ubuntu.com/ubuntu jammy InRelease
Get:5 http://us.archive.ubuntu.com/ubuntu jammy-updates InRelease [119 kB]
Hit:6 https://linux.teamviewer.com/deb stable InRelease
Ign:7 https://download.webmin.com/download/newkey/repository stable InRelease
Hit:8 https://dl.google.com/linux/chrome/deb stable InRelease
Get:9 https://pkgs.tailscale.com/stable/ubuntu jammy InRelease
Hit:10 https://download.webmin.com/download/newkey/repository stable Release
Hit:11 http://packages.azlux.fr/debian stable InRelease
Get:12 https://zotero.retorque.re/file/apt-package-archive ./ InRelease [2,153 B]
Hit:13 http://us.archive.ubuntu.com/ubuntu jammy-backports InRelease
Hit:14 https://esm.ubuntu.com/apps/ubuntu jammy-apps-security InRelease
Hit:15 https://ppa.launchpadcontent.net/agornostal/ulauncher/ubuntu jammy InRelease
Get:17 http://security.ubuntu.com/ubuntu jammy-security amd64 Contents (deb) [103 MB]
Hit:18 https://esm.ubuntu.com/apps/ubuntu jammy-apps-updates InRelease
Hit:19 https://ppa.launchpadcontent.net/andch/staging-fprint/ubuntu jammy InRelease
Hit:20 https://ppa.javinator9889.com all InRelease
Get:21 http://us.archive.ubuntu.com/ubuntu jammy-updates i386 Contents (deb) [48.8 MB]
Hit:22 https://esm.ubuntu.com/infra/ubuntu jammy-infra-security InRelease
Hit:23 https://ppa.launchpadcontent.net/christian-boxdoerfer/fsearch-stable/ubuntu jammy InRelease
Hit:24 https://esm.ubuntu.com/infra/ubuntu jammy-infra-updates InRelease
Hit:25 https://ppa.launchpadcontent.net/flexiondotorg/quickemu/ubuntu jammy InRelease
Hit:26 https://ppa.launchpadcontent.net/kubuntu-ppa/backports-extra/ubuntu jammy InRelease
Hit:27 https://ppa.launchpadcontent.net/kubuntu-ppa/backports/ubuntu jammy InRelease
Hit:28 https://ppa.launchpadcontent.net/michel-slm/distrobox/ubuntu jammy InRelease
Hit:29 https://ppa.launchpadcontent.net/uunicorn/open-fprintd/ubuntu jammy InRelease
Hit:30 https://ppa.launchpadcontent.net/yannick-mauray/quickgui/ubuntu jammy InRelease
Get:31 http://us.archive.ubuntu.com/ubuntu jammy-updates amd64 Contents (deb) [106 MB]
Get:32 http://security.ubuntu.com/ubuntu jammy-security i386 Contents (deb) [46.5 MB]
Fetched 304 MB in 1min 35s (3,204 kB/s)
Reading package lists...

```

**Comment by philclifford** [comment]
Many issues have been fixed since the 0.4.0-1 release.  A new release is awaited but in the meantime you might find that grabbing `deb-get` from the HEAD of the master branch is beneficial.  When you run `deb-get update` you should wait for the background process to build a cached list of supported apps before doing a `deb-get list`. You can keep an eye on progress using `deb-get cache`. After an install ( or adding an external repo ) you might need to do a couple of `deb-get update`s.

**Comment by chief-wrench** [comment]
Soooo what is holding up the release of 0.4.0-1? This is a pretty big issue, any progress would be appreciated. No pressure, though ;)
